### PR TITLE
Added import fixers integration test

### PR DESCRIPTION
**Pull request by GrahamCampbell**
#1424 with an integration test.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Our integration tests are a bit crap at telling us what went wrong.

**Comment by keradus** [comment]
Indeed. Feel free to enhance them!

**Comment by GrahamCampbell** [comment]
Meh. Turns out it was due to duplicating `Bar` so the lining failed.

**Comment by keradus** [comment]
> Errors reported during.

and that one msg has been already improved on 2.x line :D

**Comment by keradus** [comment]
This would be the last one into 1.10.1, when tests will pass.

**Comment by GrahamCampbell** [comment]
Sure. I need to look at why this isn't passing.

**Comment by GrahamCampbell** [comment]
Hmmm:

```
array(1) {
  [0]=>
  array(3) {
    ["type"]=>
    int(1)
    ["filepath"]=>
    string(91) "/media/sf_GitHub/GrahamForks/PHP-CS-Fixer/Symfony/CS/Tests/Fixtures/Integration/tmp/tmp.php"
    ["message"]=>
    string(3241) "exception 'RuntimeException' with message 'Index invalid or out of range' in /media/sf_GitHub/GrahamForks/PHP-CS-Fixer/Symfony/CS/Fixer/Contrib/OrderedUseFixer.php:117
Stack trace:
#0 /media/sf_GitHub/GrahamForks/PHP-CS-Fixer/Symfony/CS/Fixer/Contrib/OrderedUseFixer.php(117): Symfony\CS\Fixer\Contrib\OrderedUseFixer->getNewOrder()
#1 /media/sf_GitHub/GrahamForks/PHP-CS-Fixer/Symfony/CS/Fixer/Contrib/OrderedUseFixer.php(39): Symfony\CS\Fixer\Contrib\OrderedUseFixer->getNewOrder(Array, Object(Symfony\CS\Tokenizer\Tokens))
#2 /media/sf_GitHub/GrahamForks/PHP-CS-Fixer/Symfony/CS/Fixer.php(209): Symfony\CS\Fixer\Contrib\OrderedUseFixer->fix(Object(SplFileInfo), '<?php\n\nnamespac...')
#3 /media/sf_GitHub/GrahamForks/PHP-CS-Fixer/Symfony/CS/Tests/AbstractIntegrationTest.php(155): Symfony\CS\Fixer->fixFile(Object(SplFileInfo), Array, false, true, Object(Symfony\CS\FileCacheManager))
#4 /media/sf_GitHub/GrahamForks/PHP-CS-Fixer/Symfony/CS/Tests/AbstractIntegrationTest.php(77): Symfony\CS\Tests\AbstractIntegrationTest->doTestIntegration('imports.test', 'Integration of ...', Array, '<?php\n\nnamespac...', '<?php\n\nnamespac...')
#5 [internal function]: Symfony\CS\Tests\AbstractIntegrationTest->testIntegration('imports.test', 'Integration of ...', Array, '<?php\n\nnamespac...', '<?php\n\nnamespac...')
#6 phar:///media/sf_GitHub/GrahamForks/PHP-CS-Fixer/phpunit/phar/phpunit/Framework/TestCase.php(886): ReflectionMethod->invokeArgs(Object(Symfony\CS\Tests\IntegrationTest), Array)
#7 phar:///media/sf_GitHub/GrahamForks/PHP-CS-Fixer/phpunit/phar/phpunit/Framework/TestCase.php(746): PHPUnit_Framework_TestCase->runTest()
#8 phar:///media/sf_GitHub/GrahamForks/PHP-CS-Fixer/phpunit/phar/phpunit/Framework/TestResult.php(601): PHPUnit_Framework_TestCase->runBare()
#9 phar:///media/sf_GitHub/GrahamForks/PHP-CS-Fixer/phpunit/phar/phpunit/Framework/TestCase.php(702): PHPUnit_Framework_TestResult->run(Object(Symfony\CS\Tests\IntegrationTest))
#10 phar:///media/sf_GitHub/GrahamForks/PHP-CS-Fixer/phpunit/phar/phpunit/Framework/TestSuite.php(735): PHPUnit_Framework_TestCase->run(Object(PHPUnit_Framework_TestResult))
#11 phar:///media/sf_GitHub/GrahamForks/PHP-CS-Fixer/phpunit/phar/phpunit/Framework/TestSuite.php(735): PHPUnit_Framework_TestSuite->run(Object(PHPUnit_Framework_TestResult))
#12 phar:///media/sf_GitHub/GrahamForks/PHP-CS-Fixer/phpunit/phar/phpunit/Framework/TestSuite.php(735): PHPUnit_Framework_TestSuite->run(Object(PHPUnit_Framework_TestResult))
#13 phar:///media/sf_GitHub/GrahamForks/PHP-CS-Fixer/phpunit/phar/phpunit/Framework/TestSuite.php(735): PHPUnit_Framework_TestSuite->run(Object(PHPUnit_Framework_TestResult))
#14 phar:///media/sf_GitHub/GrahamForks/PHP-CS-Fixer/phpunit/phar/phpunit/TextUI/TestRunner.php(432): PHPUnit_Framework_TestSuite->run(Object(PHPUnit_Framework_TestResult))
#15 phar:///media/sf_GitHub/GrahamForks/PHP-CS-Fixer/phpunit/phar/phpunit/TextUI/Command.php(148): PHPUnit_TextUI_TestRunner->doRun(Object(PHPUnit_Framework_TestSuite), Array)
#16 phar:///media/sf_GitHub/GrahamForks/PHP-CS-Fixer/phpunit/phar/phpunit/TextUI/Command.php(100): PHPUnit_TextUI_Command->run(Array, true)
#17 /media/sf_GitHub/GrahamForks/PHP-CS-Fixer/phpunit(547): PHPUnit_TextUI_Command::main()
#18 {main}"
  }
}
```

**Comment by GrahamCampbell** [comment]
This looks like an actual bug in the implementation.

**Comment by GrahamCampbell** [comment]
Fixed it, ish. Basically a hack fix. Would be nice to know why that case was happening though.

**Comment by keradus** [comment]
fails on hhvm

**Comment by GrahamCampbell** [comment]
Isn't the dodgie stuff a different bug though?

**Comment by GrahamCampbell** [comment]
#1424 Fixes a bug. The broken internals were there anywway.

File: Symfony/CS/Fixer/Contrib/OrderedUseFixer.php [hunk 1]
```diff
@@ -114,6 +114,10 @@ private function getNewOrder(array $uses, Tokens $tokens)
             $index = $startIndex;
 
             while ($index <= $endIndex) {
+                if (!isset($tokens[$index])) {
```
Comment (keradus) [inline Symfony/CS/Fixer/Contrib/OrderedUseFixer.php]: please revert that

Comment (GrahamCampbell) [inline Symfony/CS/Fixer/Contrib/OrderedUseFixer.php]: sure

Comment (GrahamCampbell) [inline Symfony/CS/Fixer/Contrib/OrderedUseFixer.php]: that was always the plan i guess

**Comment by GrahamCampbell** [comment]
How does this look now?

**Comment by GrahamCampbell** [comment]
Passing. :)

**Comment by keradus** [comment]
Could be rebased on 1.11 ;)

**Comment by GrahamCampbell** [comment]
k, sec

**Comment by GrahamCampbell** [comment]
Done.

**Comment by keradus** [comment]
Thank you @GrahamCampbell.

**Pull request merged**